jwst/pipeline/calwebb_image2.py

```diff
@@ -153,7 +153,9 @@ def process_exposure_product(
         # regular 2D science image types
         if input.meta.exposure.type.upper() in self.image_exptypes and \
         len(input.data.shape) == 2:
-            self.resample(input)
+            i2d = self.resample(input)
+            if self.save_results:
+                self.save_model(i2d, suffix="i2d")
```

Comment:
Isn't there something built-in somewhere that automatically knows (via a dictionary list or something like that) what suffix to use for products coming from a particular step or pipeline? Hence I'm wondering if this explicit reference to the "i2d" suffix is necessary. I'm sure @stscieisenhamer knows the answer.

Reply:
The `i2d` suffix is only defined in `resample.cfg`.

https://github.com/spacetelescope/jwst/blob/0b595e33d332cede2a7a69500e9bcb601a151954/jwst/pipeline/resample.cfg#L11

That's the only place it is defined anywhere.  So if that config file doesn't get parsed, the default, which is `resample`, gets used.

Reply:
I guess I was thinking of the list that's in /lib/suffix.py

Reply:
`/lib/suffix` only defines known suffixes; there is no single mapping of suffixes to product results. not sure that is possible nor desired.

There is a larger architecture discussion here, but the convention seems to be to leave it up to the pipelines to define what their outputs are. Hence, the fact that the resample step defines its suffix could be a point of discussion anyways. It may have been more appropriate to not be defined there but in the `calwebb_image2.cfg` file.

As such, having the pipeline hardcode the suffix is OK.

About the code itself, the current way is OK. However, it one wanted to be a bit more "keeping with the architecture", the alternate implementation would be:
```
self.resample.save_results = self.save_results
self.resample.suffix = 'i2d'
i2d = self.resample(input)
```
This in case , if `resample` did something else that is dependent on `save_results` and its suffix, it now has all the knowledge necessary.

Reply:
Thanks Jonathan.  That's a good suggestion and is a bit cleaner.

I think having the `i2d` suffix defined in `resample.cfg` is actually not good, as it gives different behavior from running with the default config vs running with the Step class.  No other steps defined the default suffix in their config files.  `resample_spec.cfg` defines suffix to be `s2d` in a similar way.